Portfolio 1

1 a
Ved første øjekast indeholder filen:
- Survived - en kolonne der indeholder information om, hvorvidt de pågældende passagerer overlevede
Overlevet=1, Død=0 (int)
- Pclass - passagerer fordelt på klasserne 1,2 og 3 (int)
- Name - Fulde navne og titler (str)
- Sex - male eller female (str)
- Age (alder) - (float)
- Siblings/spouses aboard - liste over søskende eller partnere ombord - (int)
- Parents/Children aboard - liste over forældre eller søskende ombord - (int)
- Fare - Passagerernes henholdsvise billetpriser (float)

1 b
Jeg er ikke stødt på nogle manglende data i det pågældende datasæt, udover enkelte NaN-værdier

In [23]:
#2) 
import pandas as pd
titanic=pd.read_csv('titanic.csv')
#Jeg importerer pandas og laver en variabel 'titanic' for csv-filen


In [24]:
titanic.head()
# Og sikrer mig, at der ser fornuftigt ud

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

In [25]:
#Først og fremmest sikrer jeg mig, at der ikke er nogle null-værdier
titanic.isnull().sum()

Survived                   0
Pclass                     0
Name                       0
Sex                        0
Age                        0
Siblings/Spouses Aboard    0
Parents/Children Aboard    0
Fare                       0
dtype: int64

In [26]:
#Og så identificerer jeg typerne af hver kolonne
print(titanic.dtypes)

Survived                     int64
Pclass                       int64
Name                        object
Sex                         object
Age                        float64
Siblings/Spouses Aboard      int64
Parents/Children Aboard      int64
Fare                       float64
dtype: object


Typerne stemmer overens med hvad jeg gik ud fra i opgave 1a

In [27]:
# Jeg sikrer mig at alle 887 passagerer fra listen er repræsenterede
print(len(titanic)) # succes

# Og undersøger det summen af celler i regnearket
print(titanic.size) #Hvilket lander på 7096

887
7096


3)
Først ser vi på, hvor mange af de 887 passagerer der angiveligt overlevede

In [28]:
print(titanic['Survived'])
print(titanic['Survived'].sum())


#In order to find statistics on the passenger ages, we will run the .describe() command on the Passenger-Column

0      0
1      1
2      1
3      1
4      0
      ..
882    0
883    1
884    0
885    1
886    0
Name: Survived, Length: 887, dtype: int64
342


Med udgangspunkt i at 1=overlevet og 0=død, overlevede 342 ud af de 887 passagerer

In [29]:
titanic['Age'].describe()


count    887.000000
mean      29.471443
std       14.121908
min        0.420000
25%       20.250000
50%       28.000000
75%       38.000000
max       80.000000
Name: Age, dtype: float64

In [30]:
# Ovenstående ser vi at passager-alderen spænder mellem 0.42 og op til 80 år
# Vi kan finde median-alderen ved at kalde den direkte fra kolonnen:
titanic['Age'].median() #Som tilsyneladende er 28

28.0

In [31]:
# Lad os kigge på den gennemsnitlige billetpris
titanic['Fare'].describe() #der ligger på 32,3 pund
#Og tilsyneladende er en enkelt heldig passager kommet gratis med

count    887.00000
mean      32.30542
std       49.78204
min        0.00000
25%        7.92500
50%       14.45420
75%       31.13750
max      512.32920
Name: Fare, dtype: float64

In [32]:
# Lad os så se på, om overlevelsesraten var højst for mænd eller kvinder
titanic_pivot=titanic.pivot_table(index='Sex', columns ='Survived', values='Age', aggfunc='count')
print(titanic_pivot.head())
#Ved at lave en simpel pivot-tabel, kan vi se at overlevelsesraten var højst for kvinder

Survived    0    1
Sex               
female     81  233
male      464  109


4) Findes der personer med samme efternavn?
- Først udtrækker vi Name-kolonnen som en liste og opretter en tom liste, 'last_names' til efternavnene


In [33]:
Name=(titanic['Name'])
last_names=[]


In [34]:
# Herefter laver jeg et loop, der tager den sidste værdi (efternavnet) fra 'names' og lægger dem i listen 'last_names'
for i in range(0, len(Name)):
    last_names.append(Name[i].split()[-1])
# Og jeg tjekker at alle 887 passagerer er repræsenteret
len(last_names)


887

In [35]:
# For at finde ud af om der er nogle personer med samme efternavn, laver jeg et loop der søger efter ens værdier i 'last_names'
recurrences=([last_names.count(x) for x in last_names])
print(recurrences)

[2, 1, 1, 2, 2, 3, 1, 4, 6, 2, 2, 1, 1, 9, 1, 1, 5, 4, 3, 1, 1, 1, 1, 1, 4, 4, 1, 4, 1, 1, 1, 1, 1, 1, 3, 2, 1, 1, 3, 2, 1, 2, 3, 1, 1, 1, 1, 1, 2, 6, 1, 4, 1, 1, 1, 1, 1, 3, 6, 1, 1, 4, 6, 1, 2, 1, 1, 9, 1, 1, 6, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 4, 1, 4, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2, 1, 2, 2, 3, 4, 1, 1, 1, 1, 3, 1, 2, 1, 3, 2, 1, 1, 2, 2, 9, 3, 1, 2, 2, 2, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 3, 1, 2, 1, 1, 1, 9, 4, 3, 1, 1, 2, 1, 1, 3, 4, 1, 1, 1, 7, 1, 1, 1, 2, 6, 3, 1, 6, 1, 1, 1, 5, 6, 1, 3, 1, 4, 1, 1, 1, 7, 1, 4, 2, 1, 1, 3, 1, 3, 1, 1, 1, 1, 3, 4, 1, 1, 3, 1, 1, 1, 7, 1, 1, 1, 2, 2, 1, 1, 1, 2, 1, 1, 1, 1, 3, 1, 2, 1, 4, 1, 1, 1, 1, 3, 1, 1, 1, 1, 4, 4, 2, 1, 4, 1, 1, 1, 3, 1, 1, 2, 2, 1, 1, 2, 2, 1, 2, 2, 6, 1, 2, 1, 2, 2, 1, 1, 1, 1, 1, 4, 3, 1, 1, 1, 6, 1, 3, 1, 1, 1, 2, 1, 1, 2, 1, 1, 5, 2, 1, 2, 1, 1, 3, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 3, 1, 2, 4, 2, 6, 2, 4, 3, 1, 2, 2, 1, 2, 2, 1, 1, 4, 1, 2, 1, 2, 1, 1, 1, 1, 2, 7, 1, 1, 1, 3, 2, 2, 1, 3, 3, 2, 

In [36]:
# Hvilket (ret kaotisk) viser at adskillige efternavne gentages
# Og inden jeg foretager en bedre visualisering af gentagelser i efternavnene, sikrer jeg mig, at alle passagerer er repræsenteret i 'recurrences'
print(len(last_names)==len(recurrences))

True


In [37]:
# Jeg opretter en dataframe, der viser hvor mange passagerer der har de samme efternavne
recurring_names=pd.DataFrame({'Last Name': last_names, 'Amount':recurrences})
print(recurring_names)

     Last Name  Amount
0       Braund       2
1      Cumings       1
2    Heikkinen       1
3     Futrelle       2
4        Allen       2
..         ...     ...
882   Montvila       1
883     Graham       3
884   Johnston       2
885       Behr       1
886     Dooley       1

[887 rows x 2 columns]


In [38]:

recurring_names.nlargest(1, 'Amount')
# Og det mest populære efternavn er Andersson, hvilket hele 9 ud af de 887 passagerer hedder.

Last Name  Amount
13  Andersson       9

Opg 5 I skal nu lave en pivot-tabel som viser hvor mange der rejste på hhv 1., 2. og 3. klasse. Hvilken rejseklasse havde flest omkomne?

In [39]:
# jeg laver et pivot table over, hvor mange der rejste på hver klasse. Jeg bruger 'Age' til at vise mine værdier
passenger_per_class=titanic.pivot_table(columns='Pclass', values='Age', aggfunc='count')
print(passenger_per_class)

Pclass    1    2    3
Age     216  184  487


Ovenstående ses, hvordan de 887 passagerer var fordelt på de tre klasser.

Hvilken rejseklasse havde de flest omkomne?


In [40]:
# jeg laver en ny pivot tabel, der kan vise hvor mange der henholdsvis døde og overlevede på hver klasse
pivot_passenger_class=titanic.pivot_table(index='Pclass', columns='Survived',values='Age', aggfunc='count')
print(pivot_passenger_class)

Survived    0    1
Pclass            
1          80  136
2          97   87
3         368  119


3. Klasse havde langt fest omkomne.